include/targets/cuda/reduce_helper.h

```diff
@@ -76,6 +76,7 @@ namespace quda
     cuda::atomic<system_atomic_t, cuda::thread_scope_system> *result_h; /** host atomic buffer */
     count_t *count; /** count array that is used to track the number of completed thread blocks at a given batch index */
     bool consumed; // check to ensure that we don't complete more than once unless we explicitly reset
+    T *device_output_buffer = nullptr; // Optional device output buffer for the reduction result
```

Comment:
Changes to this file must also be propagated to include/targets/generic/reduce_helper.h (which is the stream synchronized reduction, as opposed to this which is the `std::atomic` based implementation).

Reply:
https://github.com/lattice/quda/pull/1061/commits/d9e668e4f01f639c0fb01188d4fbbcd2cbf34544